api/api.go

```diff
@@ -0,0 +1,183 @@
+package api
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// NextDate function calculates the date of the next task execution based on the value in the repeat string
+// NextDate функция рассчитывает дату следующего выполнения задания исходя из значения в строке повтор
+func NextDate(now time.Time, date string, repeat string) (string, error) {
+	// если правила повторения не указано, то возвращаем пустую строку
+	if repeat == "" {
+		return "", nil
+	}
+	dateTask, err := time.Parse("20060102", date)
+	if err != nil {
+		return "", fmt.Errorf("failed to convert string to date", err)
```

Comment:
Хорошая практика описывать ошибки, только нужно добавить саму ошибку в форматирование
```suggestion
		return "", fmt.Errorf("failed to convert string to date: %w", err)
```

Reply:
Проверь по всему проекту